src/rust/engine/process_execution/src/lib.rs

```diff
@@ -539,6 +539,10 @@ pub struct Process {
 
   /// The docker image to run the process with.
   pub docker_image: Option<String>,
+
+  /// Properties used for remote execution configuration. Regardless of remote execution, these
+  /// should impact the cache key.
+  pub platform_properties: Vec<(String, String)>,
```

Comment:
If this is on the `Process` struct directly, then I don't think that it needs to be in the `ProcessMetadata`.

In REAPI, the `Platform` is a property of the `Action`, which (together with the `Command`) corresponds with our `Process` struct: https://github.com/pantsbuild/pants/blob/2a252a3c1b17c75315b01ff964cf3502709cd782/src/rust/engine/protos/protos/bazelbuild_remote-apis/build/bazel/remote/execution/v2/remote_execution.proto#L496-L505

On the other hand, the [`Action.salt` field](https://github.com/pantsbuild/pants/blob/2a252a3c1b17c75315b01ff964cf3502709cd782/src/rust/engine/protos/protos/bazelbuild_remote-apis/build/bazel/remote/execution/v2/remote_execution.proto#L489-L495) is effectively `cache_key_gen_version`, and that probably should stay in the `ProcessMetadata`.

Reply:
True. At that point, maybe `ProcessMetadata` isn't useful anymore? Note that per the above, we don't think `cache.rs` needed to be storing `instance_name` anymore.

I'll try something like this out.